csharp/src/Ice/SlicStream.cs

```diff
@@ -318,44 +281,21 @@ internal override void ReceivedReset(long errorCode)
 
         private bool ReleaseFlowControlCredit(bool notifyPeer = false)
         {
-            if (IsControl)
-            {
-                return true;
-            }
-
             // If the flow control credit is not already released, decreases the bidirectional or unidirectional
             // semaphore (for outgoing streams) / count (for incoming streams).
```

Comment:
I think we should update the comment as this no longer use the semaphore directly